p2p/libp2p/netMessenger.go

```diff
@@ -1089,9 +1093,11 @@ func (netMes *networkMessenger) UnregisterAllMessageProcessors() error {
 	defer netMes.mutTopics.Unlock()
 
 	for topic := range netMes.processors {
-		err := netMes.pb.UnregisterTopicValidator(topic)
```

Comment:
maybe refactor this way?
```GO
for topic := range netMes.processors {
    if topic == common.ConnectionTopic {
        continue
    }

    err := netMes.pb.UnregisterTopicValidator(topic)
    if err != nil {
        return err
    }

    delete(netMes.processors, topic)
}
```

Reply:
added this solution, as we still need to delete the processor from netMes.processors(saved for `directMessageHandler`)

```go
for topic := range netMes.processors {
	if topic == common.ConnectionTopic {
		delete(netMes.processors, topic)
		continue
	}

	err := netMes.pb.UnregisterTopicValidator(topic)
	if err != nil {
		return err
	}

	delete(netMes.processors, topic)
}
```